src/store.cc

```diff
@@ -1009,6 +1005,10 @@ StoreEntry::checkCachable()
         } else if (fdNFree() < RESERVED_FD) {
             debugs(20, 2, "StoreEntry::checkCachable: NO: too many FD's open");
             ++store_check_cachable_hist.no.too_many_open_fds;
+        } else if (EBIT_TEST(flags, ENTRY_NEGCACHED)) {
+            debugs(20, 3, "StoreEntry::checkCachable: YES: negative cached");
+            ++store_check_cachable_hist.yes.negative_cached;
+            return 1;
```

Comment:
Assuming the PR code is correct, this `if` statement looks out of place (at best) or even unnecessary:

1. Its addition does not change the method outcome. Its only effect is a side-effect.

2. Its only justification could be the collection of statistics about negatively cached objects (e.g., what percentage of cached objects were negatively cached), but this method is the wrong place to collect such statistics -- this method does not (fully) determine which objects will be cached. The statistics in this method is supposed to be about key decisions made in this method, but, per item 1 above, this `if` statement only pretends to make a decision.

Assuming the PR code is correct, I suggest removing this if statement and the corresponding statistics. If somebody wants to add a new "the number/ratio of cached objects that were negatively cached" and/or a new "the number/ratio of responses that were eligible for negative caching" statistics, they can certainly do that. If you want to add that statistics, you can do that in this PR, but I do not think it is _required_ for this PR acceptance. I have not researched the best place(s) to add such statistics -- it may not be easy to find the right place(s).

Reply:
The yes.negative_cached stat and corresponding code is removed.